pkg/serverless/invocationlifecycle/lifecycle_test.go

```diff
@@ -183,3 +184,98 @@ func TestEndExecutionSpanWithLambdaLibrary(t *testing.T) {
 
 	assert.Equal(t, (*api.Payload)(nil), tracePayload)
 }
+
+func TestCompleteInferredSpanWithStartTime(t *testing.T) {
+	defer os.Unsetenv(functionNameEnvVar)
+	os.Setenv(functionNameEnvVar, "TestFunction")
+
+	extraTags := &logs.Tags{
+		Tags: []string{"functionname:test-function"},
+	}
+	demux := aggregator.InitTestAgentDemultiplexerWithFlushInterval(time.Hour)
+	mockDetectLambdaLibrary := func() bool { return false }
+
+	var tracePayload *api.Payload
+	mockProcessTrace := func(payload *api.Payload) {
+		tracePayload = payload
+	}
+	startInferredSpan := time.Now()
+	time.Sleep(250 * time.Millisecond)
```

Comment:
this will slow down the test, maybe you can compute the current time + 250ms without having to wait for it